CRM/Admin/Form/Setting/Debugging.php

```diff
@@ -54,8 +54,9 @@ public function buildQuickForm() {
     }
 
     parent::buildQuickForm();
-
-    if (defined('CIVICRM_ENVIRONMENT')) {
+    Civi::service('settings_manager')->useMandatory();
+    $manualSetting = Civi::settings()->get('environmentManual');
+    if ($manualSetting) {
```

Comment:
@seamuslee001 understood that the goal is to disable GUI elements if the option has been manually set.

The additional setting `environmentManual` feels unnecessary. To see if the sysadmin has done an override, try calling the `getMandatory()` function. 

```php
if (Civi::service()->getMandatory('environment') !== NULL) {
```

If the `environment` setting has been set as a mandatory value (via `$civicrm_settings`), then `getMandatory()` should return that value. Otherwise, it should return `NULL`.

Reply:
@seamuslee001 I think this should go against the rc once you have processed Tim's comment